rust/src/mqtt/mqtt.rs

```diff
@@ -462,13 +462,13 @@ impl MQTTState {
         }
 
         while !current.is_empty() {
-            SCLogDebug!("request: handling {}", current.len());
+            SCLogDebug!("request: handling {}; rem: {}", current.len(), input.len() - current.len());
             match parse_message(current, self.protocol_version, self.max_msg_len) {
                 Ok((rem, msg)) => {
                     let _pdu = Frame::new(
                         flow,
                         &stream_slice,
-                        input,
+                        current,
                         current.len() as i64,
```

Comment:
Should it not be `current.len()-rem.len()` ? (and same below)

Reply:
Will adjust in new pr